packages/graphql/src/lib.ts

```diff
@@ -93,15 +97,30 @@ const EmitterOptionsSchema: JSONSchemaType<GraphQLEmitterOptions> = {
 
 export const libDef = {
   name: "@typespec/graphql",
-  diagnostics: {},
+  diagnostics: {
+    "graphql-operation-kind-duplicate": {
+      severity: "error",
+      messages: {
+        default: paramMessage`GraphQL Operation Kind already applied to \`${"entityName"}\`.`,
+      },
+    },
+  },
   emitter: {
     options: EmitterOptionsSchema as JSONSchemaType<GraphQLEmitterOptions>,
   },
   state: {
+    operationKind: {
+      description:
+        "State for the graphql operation kind decorators (@query, @mutation, @subscription)",
+    },
     schema: { description: "State for the @schema decorator." },
   },
 } as const;
 
 export const $lib = createTypeSpecLibrary(libDef);
 
-export const { reportDiagnostic, createDiagnostic, stateKeys: GraphQLKeys } = $lib;
+export const {
+  reportDiagnostic,
+  createDiagnostic,
+  stateKeys: GraphQLKeys,
+} = $lib;
```

Comment:
Are you using eslint/prettier as defined in the TypeSpec repo? We should be sure to follow the same style rules as upstream.